### PR TITLE
Add kustomization manifests to CRDs and controller components

### LINKED ISSUE #605: Kustomize manifests for CRDs and snapshot controller?
This repository is not very GitOps-friendly. While building and maintaining a Helm chart as suggested in #551 might be something that's not very high on the priority list, adding `kustomization.yaml` files to a few directories is neither a huge task nor does it require maintenance, and it would (at least for me and likely other users of automation frameworks like FluxCD) pretty much achieve the same thing - being able to install CRDs and controller from a versioned git repository.

I'll open a PR implementing this shortly.

### PR DESCRIPTION
**Pull request by itspngu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind design

**What this PR does / why we need it**:

The changes introduced with this PR allow for better integration of the controller components and, in particular, the CRDs (which other projects rely on) into a GitOps workflow by providing a means to reference versioned manifests via `kustomize`.

**Which issue(s) this PR fixes**:

Fixes #605 

**Special notes for your reviewer**:

none

**Does this PR introduce a user-facing change?**:

```release-note
Added kustomization manifests to CRDs and controller components
```

### CODE REVIEW

File: client/config/crd/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - snapshot.storage.k8s.io_volumesnapshotclasses.yaml
+  - snapshot.storage.k8s.io_volumesnapshotcontents.yaml
+  - snapshot.storage.k8s.io_volumesnapshots.yaml
```
Comment (xing-yang) [inline client/config/crd/kustomization.yaml:6]: @pohly Do you see any issue with this change?  We install snapshot CRDs from external-snapshotter/pkg/client/config/crd folder:
https://github.com/kubernetes-csi/csi-release-tools/blob/master/prow.sh#L742

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 2]
```diff
@@ -118,4 +118,4 @@ spec:
               mountPath: /csi
       volumes:
         - name: socket-dir
-          emptyDir:
+          emptyDir: {}
```
Comment (pohly) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:121]: Why is this needed?

File: client/config/crd/kustomization.yaml (hunk 1 above)
Comment (pohly) [reply client/config/crd/kustomization.yaml:6]: Adding the kustomization.yaml makes sense to me, I can imagine that some users may want to use the directory as their base.

But what does that have to do with prow.sh? Our usage of the files will simply ignore the additional kustomization.yaml.

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml (hunk 2 above)
Comment (itspngu) [reply deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:121]: VSCode was complaining about it (`Incorrect type. Expected "object".`) when I went through the files and I must have impulsively... "fixed" it. 

To be on the safe side, I just ran kubeval on it. It doesn't mind either version (with or without empty object). I can remove that bit if you'd prefer to keep it as is to avoid complications in automation, sorry for not noticing myself when opening the PR.

Comment (pohly) [reply deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:121]: Let's keep it. I was just wondering because it seemed unrelated to the stated purpose of the PR.

File: client/config/crd/kustomization.yaml (hunk 1 above)
Comment (xing-yang) [reply client/config/crd/kustomization.yaml:6]: Just want to double check.  It should be fine.

Comment (xing-yang) [reply client/config/crd/kustomization.yaml:6]: @itspngu Can you update the "Install Snapshot CRDs:" section from "kubectl create -f client/config/crd" to install individual CRDs?  I want to avoid confusion for folks who are new to this.
https://github.com/kubernetes-csi/external-snapshotter/blob/master/README.md#usage

Comment (itspngu) [reply client/config/crd/kustomization.yaml:6]: > install individual CRDs

Good catch! Are you sure you want all the individual files though? That'd be rather verbose. Part of the beauty of kustomize is that you can easily do something like this:

```
$ kubectl kustomize client/config/crd | kubectl create -f - --dry-run=server
Error from server (AlreadyExists): error when creating "STDIN": customresourcedefinitions.apiextensions.k8s.io "volumesnapshotclasses.snapshot.storage.k8s.io" already exists
Error from server (AlreadyExists): error when creating "STDIN": customresourcedefinitions.apiextensions.k8s.io "volumesnapshotcontents.snapshot.storage.k8s.io" already exists
Error from server (AlreadyExists): error when creating "STDIN": customresourcedefinitions.apiextensions.k8s.io "volumesnapshots.snapshot.storage.k8s.io" already exists
```

As such, I'd update the readme with something similar to the above where needed, unless you prefer the explicit way using individual files.

Comment (xing-yang) [reply client/config/crd/kustomization.yaml:6]: That should be fine.  Thanks

Comment (itspngu) [reply client/config/crd/kustomization.yaml:6]: I pushed an update to the readme. Let me know if there's anything else to change.

**Comment by xing-yang** [comment]
@itspngu Can you backport this to release-5.0 branch?

**Comment by itspngu** [comment]
> @itspngu Can you backport this to release-5.0 branch?

Okay, will do so later today in a separate PR.
